paasta_tools/marathon_tools.py

```diff
@@ -326,8 +327,21 @@ def format_marathon_app_dict(self, app_id, docker_url, docker_volumes, service_n
                                marathon configuration file
         :param service_namespace_config: The service instance's configuration dict
         :returns: A dict containing all of the keys listed above"""
+        instances = self.get_instances()
+
+        # A set of config attributes that don't get included in the hash of the config.
+        # These should be things that PaaSTA/Marathon knows how to change without requiring a bounce.
+        CONFIG_HASH_BLACKLIST = set(['instances', 'backoff_seconds'])
+
+        system_paasta_config = load_system_paasta_config()
+        docker_url = get_docker_url(system_paasta_config.get_docker_registry(), self.get_docker_image())
+        service_namespace_config = load_service_namespace_config(
```

Comment:
We are losing something here, and that is a soa_dir arg. :(

I think if you are going to do this kind of change where the object itself is responsible for reading in more config files, you are going to have to make the init take in the soa_dir arg to let it know "where it came from"